src/Factory/AdapterFactory.php

```diff
@@ -224,4 +231,10 @@ private function guardFactoryClassImplementation($adapterFactory, $label)
             self::SUPPORT_ISSUE_TRACKER => $issueTracker,
         ];
     }
+
+    private  function getBaseAdapter($name) {
+      $array = explode(':', $name);
+      return $array[0];
+    }
+
```

Comment:
we are not concerned about BC at all, gush is a tool and we can break BC anytime needed.

At least for now, it is stable but if this is needed we can do it without problems, ping @sstok